### ISSUE TITLE
JSON-style hash keys don't work with array/hash as a value

### ISSUE THREAD

**Issue by jgaskins** [opal/opal#1490]
Ruby supports a JSON-style syntax for hashes:

```ruby
    {
      "display": "inline-block",
    }
```

This is awesome for copy/pasting JSON blobs into Ruby and it works just as expected. And this works in Opal just fine. However, when I tried to do that using an array as a value:

```ruby
    {
      "font-family": ['Helvetica Neue', 'sans-serif'],
    }
```

I got this error:

```
SyntaxError: An error occurred while compiling: app

parse error on value "[" (tLBRACK2) :app:52
Source: app:52:24
        "font-family": ['Helvetica Neue', 'sans-serif'],
~~~~~~~~~~~~~~~~~~~~~~~^
```

Same thing when I tried it using a `"style"` key to point to this hash as its value. This is all on `0.10.0.rc2`. I want to say this used to work (I do all my Clearwater app styles in Ruby hashes, so I thought I'd done this before). Could this be due to the move to `ruby-parser`? Is that included in the rc2 release?

**Comment by elia** [comment]
> Could this be due to the move to ruby-parser? Is that included in the rc2 release?

No, I froze 0.10 just before that merge, `parser` will be part of 0.11, hopefully soon after 0.10 is out.

@jgaskins I tried a couple versions and I get the same error on 0.9.4, 0.10.beta1, 0.10.beta3 and 0.10.beta5, unsurprisingly it works fine on master (thanks to `parser`).

I'll close this as AFAICT it's not a regression, but I'd be ok with a PR fixing it on the `0-10-stable` and subsequently releasing it as 0.10.1.

**Comment by jgaskins** [comment]
Ah, cool. Looks like I was mistaken about it working before. I guess I'd only tried it with string values. Glad it's working now, though!

**Comment by jgaskins** [comment]
Oh wait, it's not that it'd be a regression, but that it doesn't match Ruby behavior.

**Comment by elia** [comment]
@jgaskins yeah, the point about that being a regression was really about releases. That is that it's not going to be fixed/released in 0.10.0 and it's already fixed on master (will be 0.11).

**Comment by jgaskins** [comment]
Ah, right. My bad.
